### DISCUSSION TITLE
Weird https page

### DISCUSSION THREAD

**Discussion by zarevskaya** [JulianPrieber/llc-docker#61] [Issues]
Hi,

I use Linkstack since 1 day and I have an issue with https.

Platform: Debian 11, Linkstack on Docker

![image](https://github.com/LinkStackOrg/LinkStack/assets/60230221/1262be78-ff20-41db-9451-332aaee1ebd4)
![Screenshot 2023-05-18 at 12-34-39 MMlink](https://github.com/LinkStackOrg/LinkStack/assets/60230221/2c3a9836-2a85-4de4-b3ea-51c16e230212)

http://" which tries to force the links to be HTTP even if the site is sent in HTTPS

Thank you :)

**Comment by JulianPrieber** [comment]
This error is widely documented here.
Please use HTTPS for your reverse proxy, not HTTP as stated in the docs.
You can try to turn on force HTTPS in your config and see if that fixes it.

**Comment by zarevskaya** [comment]
Thank you, I check again :)

**Comment by azukaar** [comment]
@JulianPrieber they are using HTTPS in the reverse proxy, but are getting "mixed content" errors because linkstack continues to send HTTP files over HTTPS.

I am the dev of the reverse proxy they are using, how does linkstack decides what protocol to use? May be the proxy is not sending a header that linkstack expects?

**Comment by JulianPrieber** [comment]
LinkStack doesn't really 'decide' on what protocol to use. It simply responds in whatever protocol it's accessed with.
If you access the page over something like `proxy_pass http://127.0.0.1:80;` all URLs will be generated as HTTP.

**Comment by azukaar** [comment]
Ah so you are deciding depending on the inbound request? Isn't that going to cause an issue with every reverse proxy unless you use HTTPS between the reverse proxy and LinkStack, no? Or may be I misunderstood.

If that's the case, with your answer I am assuming that this is not the first time you see the issue. I am assuming that you have a reason not to use relative path (which would solve the issue already). So one easy solution is to set the links to start with `//`. If you do this, the browser itself is going to decide whether to use HTTP or HTTPS, so even if Linkstack is served with

User -> HTTPS -> Proxy -> HTTP -> LinkStack

Then the browser will simply use HTTPS to request the files (and the proxy will translate to HTTP)

Something like 

```
<script src="//mydomain.com/myfile.js"
```

Doc here: http://www.ietf.org/rfc/rfc3986.txt

**Comment by JulianPrieber** [comment]
That's right. 
Relative paths aren't currently viable for us. Maybe in the future we can work something out, but as of now we have to let the user use this specific setup to avoid mixed content errors.

**Comment by azukaar** [comment]
Note that starting  with  // are still absolute path, but they will fix your issue for good with mixed contents

**Comment by JulianPrieber** [comment]
Unfortunately, generating relative URLs without including the protocol directly through Laravel's built-in methods may not be a straightforward task in our current system. The application is a fork of an existing project that follows the standard Laravel approach for URL generation.

Rewriting the entire application to accommodate relative URLs would require significant effort and resources. Given the scope of the task and the constraints, it may not be feasible to pursue this solution at the moment.

However, if someone else is interested in implementing this functionality and can contribute to the project, I would be open to including their work.
